### DISCUSSION TITLE
Require Blog theme in Razor Syntax instead of Liquid

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#12477] [Q&A]
Folks, Currently The blog theme is in liquid 

![image](https://user-images.githubusercontent.com/5202291/191838936-f3295385-f7b5-43f7-a737-066caf10be1d.png)

we require Blog theme in Razor Syntax instead of Liquid is there any example or source so please share for understanding the functionality easily. Please advise @ns8482e @Skrypt

**Comment by Skrypt** [comment]
We don't have this because most people will use these as Liquid templates among other Razor templates. Else, you will need to translate them simply.

**Reply by saleemhaq** [reply]
> We don't have this because most people will use these as Liquid templates among other Razor templates. Else, you will need to translate them simply.

How can I open option to put comments like below:

![image](https://user-images.githubusercontent.com/5202291/191945608-302ffd84-8c2e-4427-a96f-640635aa215c.png)

of, site user and publically is visible for all, currently i see on blog theme there is no option for put comments:

PLEASE advise @Skrypt @ns8482e

**Reply by ns8482e** [reply]
If you know ASP.NET Razor views syntax then it’s really easy to translate  all blog theme liquid pages to cshtml

- convert all `{{ XYZ }}` to `@Model.XYZ` 
- keep file name same and change ext put them in Views folder and inherit  orchard’s RazorPage
- Model you will get in pages are dynamic so use Orchard.ConsoleLog to find properties that you can use
- Razor Tag helpers are available same as in liquid
- Few logic  like variables/control flow unique to liquid need to be converted to c# , you may need to learn liquid syntax to understand current blog theme logic

**Reply by saleemhaq** [reply]
> If you know ASP.NET Razor views syntax then it’s really easy to translate all blog theme liquid pages to cshtml
> 
> * convert all `{{ XYZ }}` to `@Model.XYZ`
> * keep file name same and change ext put them in Views folder and inherit  orchard’s RazorPage
> * Model you will get in pages are dynamic so use Orchard.ConsoleLog to find properties that you can use
> * Razor Tag helpers are available same as in liquid
> * Few logic  like variables/control flow unique to liquid need to be converted to c# , you may need to learn liquid syntax to understand current blog theme logic

I mean to say I create blog post from orchard side and when blogpost published every blog post has this option, how i put that option? so i put that comment option to all blog post and showing in trail as well currently orchard has this option of commenting on blogpost ? please advise @ns8482e
